src/amber/run_macros/generate_asset_helpers.cr

```diff
@@ -0,0 +1,58 @@
+require "json"
+require "colorize"
+
+class AssetManifestBuilder
+  MANIFEST_PATH = File.expand_path("./config/webpack/manifest.json")
+  MAX_RETRIES   =   20
+  RETRY_AFTER   = 0.25
+
+  property retries
+  @retries : Int32 = 0
+
+  def self.build_with_retry
+    new.build_with_retry
+  end
+
+  def build_with_retry
+    if manifest_exists?
+      build
+    else
+      retry_or_raise_error
+    end
+  end
+
+  private def retry_or_raise_error
+    if retries < MAX_RETRIES
+      self.retries += 1
+      sleep(RETRY_AFTER)
+      build_with_retry
+    else
+      raise_missing_manifest_error
+    end
+  end
+
+  private def build
+    manifest_file = File.read(MANIFEST_PATH)
+    manifest = JSON.parse(manifest_file)
+
+    manifest.each do |key, value|
+      puts %({% ASSET_MANIFEST["#{key.as_s}"] = "#{value.as_s}" %})
+    end
+  end
+
+  private def manifest_exists?
+    File.exists?(MANIFEST_PATH)
+  end
+
+  private def raise_missing_manifest_error
+    puts "Manifest at #{AssetManifestBuilder::MANIFEST_PATH} does not exist".colorize(:red)
+    puts "Make sure you have compiled your assets".colorize(:red)
+  end
+end
+
+begin
+  AssetManifestBuilder.build_with_retry
```

Comment:
Just to make sure I understand the flow here:

- Developer runs `crystal run src/app.cr` (or other build command)
- If assets aren't built, this script will wait up to `MAX_RETRIES`x`RETRY_AFTER`=5s for them
- If no assets are present still, this will abort the compile
- Otherwise, this provides a compiled list, provided by webpack, of static assets

Is that right?

Reply:
Essentially yes.  And I have to give credit for this stuff to Lucky Framework.  But they seemed to borrow heavily from Amber, so I think it's fair :).